### ISSUE TITLE
API quirk: Accessing configuration during building services is awkward

### ISSUE THREAD

**Issue by davidfowl** [davidfowl/FeatherHttp#12]
Usually in ASP.NET Core, `IConfiguration` is available during ConfigureServices to allow configuring things like connection strings etc. It's currently impossible to do the same thing with feather e.g.:

```C#
WebApplication app = null;

builder.Services.AddSignalR().AddStackExchangeRedis(o =>
{
    // Read the redis service connection string
    var connection = app.Configuration["Redis:ConnectionString"];
    o.Configuration.EndPoints.Add(connection);
});
```

The above is getting a connection string from configuration and this only works because there happens to be a lambda that can also configure `AddStackExchangeRedis`. Now, most of the APIs in ASP.NET Core have a lambda version but it's really hard to use the version that takes a connection string directly:

```C#
var connectionString = ??;
builder.Services.AddSignalR().AddStackExchangeRedis(connectionString);
```

Since we haven't built configuration as yet it's not possible to get the connection string this early.

**Comment by davidfowl** [comment]
Fixed with #20

**Comment by davidfowl** [comment]
```C#
var connectionString = builder.Configuration["Redis:ConnectionString"];
builder.Services.AddSignalR().AddStackExchangeRedis(connectionString);
```
